lib/process-services-cloud/src/lib/form/components/widgets/radio-buttons/radio-buttons-cloud.widget.spec.ts

```diff
@@ -172,4 +173,25 @@ describe('RadioButtonsCloudWidgetComponent', () => {
         expect(selectedOption.innerText).toBe('opt-name-1');
         expect(widget.field.isValid).toBe(true);
     });
+
+    it('should show error message if the restUrl failed to fetch options', () => {
+        const jsonDataSpy = spyOn(formCloudService, 'getRestWidgetData').and.returnValue(throwError('Failed to fetch options'));
+        widget.field.restUrl = 'https://fake-rest-url';
+        widget.field.optionType = 'rest';
+        widget.field.restIdProperty = 'name';
+        fixture.detectChanges();
+
+        const radioButtons = element.querySelector<HTMLInputElement>('mat-radio-group');
+        radioButtons.click();
+        fixture.detectChanges();
+
+        const errorMessage = element.querySelector('.adf-radio-button-failed-rest-api-message .adf-error-text');
+        const errorIcon = element.querySelector('.adf-radio-button-failed-rest-api-message .adf-error-icon');
+
+        expect(jsonDataSpy).toHaveBeenCalled();
+        expect(widget.isRestApiFailed).toBe(true);
+        expect(widget.field.options.length).toEqual(0);
+        expect(errorIcon.textContent).toBe('error_outline');
```

Comment:
I think we should just check the error is displayed not also all the logic behind it...

we can remove 
 expect(jsonDataSpy).toHaveBeenCalled();
        expect(widget.isRestApiFailed).toBe(true);
        expect(widget.field.options.length).toEqual(0);

Reply:
removed